src/graph/graph.h

```diff
@@ -25,6 +25,17 @@
 #define GRAPH_NO_RELATION -1                 // Relations are numbered [0-N], -1 represents no relation.
 #define GRAPH_UNKNOWN_RELATION -2            // Relations are numbered [0-N], -2 represents an unknown relation.
 
+// Mask with most significat bit on 10000....
+#define MSB_MASK 1UL << ((sizeof(EntityID) * 8) - 1)
+// Set X's most significat bit on.
+#define SET_MSB(x) x | MSB_MASK
+// Clear X's most significat bit on.
+#define CLEAR_MSB(x) x ^ MSB_MASK
```

Comment:
This toggles the MSB, rather than clearing it. To clear it, do something like:
`x & ~MSB_MASK`
We currently only get to this point after an `if (SINGLE_EDGE(x))` check, so right now there is no functional difference.